otelcol/flags.go

```diff
@@ -50,9 +51,12 @@ func flags(reg *featuregate.Registry) *flag.FlagSet {
 			return nil
 		})
 
-	flagSet.Var(featuregate.NewFlag(reg), featureGatesFlag,
-		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.")
+	gatesFlag, strictFlag := featuregate.NewFlags(reg)
+	flagSet.Var(strictFlag, featureGatesStrictFlag,
```

Comment:
Instead of returning the `strictFlag` which also breaks the API for the featuregate why not adding it as as flag to the `gatesFlag` set?

Reply:
It doesn't change much. It pretty much takes this code and moves it inside the featuregate package I guess. If that's what we want to do, sure.

Reply:
I think I was confused by the API, my proposal doesn't make sense. But I still don't like the new API. 

For me this is an option/property of the `featuregate.NewFlag` so, let's change the old API to `func NewFlag(reg *Registry, opts ...FlagOption) flag.Value` and pass the strictness as option.

Reply:
strictness is not an option passed in, it's a flag. This API creates flags for the user to fill in.

Reply:
Where do we go from here? I am not sure how to address this feedback. I can bring the method `NewFlag` to deprecate it.

Reply:
> For me this is an option/property of the `featuregate.NewFlag` so, let's change the old API to `func NewFlag(reg *Registry, opts ...FlagOption) flag.Value` and pass the strictness as option.

This doesn't make sense to me. The PR proposes to add the concept of strictness as a user-defined, runtime option. If I'm understanding @bogdandrutu's suggestion correctly, it would have to be a a developer-defined compile-time option, which would serve a very different purpose.